doc/COMPILE.md

```diff
@@ -321,20 +318,26 @@ The SWIG executable may be specified with:
 
 If this option is not used, cmake will find the first occurrence of
 `swig` in your environment's path.
-Even with `ALL` GI bindings (deprecated) and gsettings (experimental) are not included.
-To include them, use:
+To build GI bindings (deprecated) and gsettings (experimental) use:
 
-    -DBINDINGS="ALL;GI;gsettings"
+    -DBINDINGS="GI;gsettings"
 
 Some bindings provide different APIs (and not a different language), e.g:
 
 - `gsettings`
 - `INTERCEPT` with `intercept_fs` and `intercept_env`
+- `IO` with `io_uv`
 
 To not add such APIs, but only `swig` bindings and `cpp`, you can use:
 
     -DBINDINGS="SWIG;cpp"
 
+Other available categories of bindings are:
+
+    - `DEFAULT` to build default bindings
+		- `STABLE` to build all stable bindings (without experimental and deprecated
```

Comment:
Ah its not subcategory the identation is a formatting error I missed.

I don't mind changing it. I tried to replicate the previous behavior where possible:

- DEFAULT: cpp (was used when nothing was passed for BINDINGS)
- STABLE (former ALL): cpp, jna, glib, SWIG, INTERCEPT
- ALL: now every binding
- EXPERIMENTAL: gsettings, haskell, io_uv
- DEPRECATED: GI

"ALL;-EXPERIMENTAL" is good for consistency, but it includes more bindings than the previous default (and deprecated bindings).

Reply:
Thank you. STABLE is, however, not listed in `doc/CONTRACT.ini`? If it is only a synonym for `ALL;-EXPERIMENTAL;-DEPRECATED` we do not need it or did you have something else in mind?

jna is actually experimental, it was only an oversight in the last release. intercept_fs also is experimental (see at the end of README.md)

glib/SWIG is currently unmaintained (I will ask @manuelm about them).

So maybe we make the default to be `MAINTAINED;-EXPERIMENTAL;-DEPRECATED` (for both plugins and bindings). DEFAULT would not be needed then. Any opinions on that? (@sanssecours ?)

@waht If the discussion here gets to long, please move such proposals to a separate issue.

Reply:
> So maybe we make the default to be `MAINTAINED;-EXPERIMENTAL;-DEPRECATED` (for both plugins and bindings). 

Sounds good to me.

> DEFAULT would not be needed then. Any opinions on that?

I like the addition of `DEPRECATED`. I do not think we need to add `STABLE` or `DEFAULT`, since we can use `MAINTAINED;-EXPERIMENTAL;-DEPRECATED` and `cpp` instead.

Reply:
> Thank you. STABLE is, however, not listed in `doc/CONTRACT.ini`? If it is only a synonym for `ALL;-EXPERIMENTAL;-DEPRECATED` we do not need it or did you have something else in mind?

I did not think of the statuses in `CONTRACTS.ini` but it makes sense to apply them to bindings as well. Yes, `STABLE` it would be just a synomym.
I'll update the usedBy in `CONTRACTS.ini`.

> I like the addition of DEPRECATED. I do not think we need to add `STABLE` or `DEFAULT`, since we can use `MAINTAINED;-EXPERIMENTAL;-DEPRECATED` and `cpp` instead.

Ok, great. I'll update the build scripts to have either no value (the new default `MAINTAINED;-EXPERIMENTAL;-DEPRECATED`) or where currently no value is present `cpp`. This will keep the build process reasonably quick.

> @waht If the discussion here gets to long, please move such proposals to a separate issue.

The BINDINGS overhaul got a bit more thorough than I expected :)